extensions/guacamole-auth-ldap/src/main/java/org/apache/guacamole/auth/ldap/AuthenticationProviderService.java

```diff
@@ -308,25 +316,49 @@ public LDAPAuthenticatedUser authenticateUser(Credentials credentials)
     }
 
     /**
-     * Returns parameter tokens generated from LDAP attributes on the user
-     * currently bound under the given LDAP connection. The attributes to be
-     * converted into parameter tokens must be explicitly listed in
-     * guacamole.properties. If no attributes are specified or none are
-     * found on the LDAP user object, an empty map is returned.
+     * Returns the current LDAP domain token from the provided user credentials
+     * @param credentials
+     *     The credentials used for authentication.
+     *
+     * @return
+     *     Domain name by splitting login username or null if no domain is detected.
+     */
+    private String getDomainToken(Credentials credentials) {
+        String username = credentials.getUsername();
+        Pattern pattern = Pattern.compile("^(.+)\\\\.*$|^.*@(.+)$");
```

Comment:
Might be nice to document what you're doing, here - regular expressions can be obscure enough that it isn't obvious what this RegEx does, and would benefit from a one line description of it.